### PR TITLE
Fix ovs-testcontroller build and address other CMake issues

### PR DESCRIPTION
**Pull request by ffoulkes**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,9 @@ project(networking-recipe VERSION 0.1 LANGUAGES C CXX)
 include(FindPkgConfig)
 include(CMakePrintHelpers)
 
+# Default: Release with Debug Info
+set(CMAKE_BUILD_TYPE "RelWithDebInfo")
```
Comment (nupuruttarwar) [inline CMakeLists.txt:15]: RelWithDbgInfo?

**Review by nupuruttarwar — APPROVED** [review]
In general LGTM

File: CMakeLists.txt (hunk 1 above)
Comment (ffoulkes) [reply CMakeLists.txt:15]: Release with Debug Information. It sets `-O2 -g -DNDEBUG`.
Release sets `-O3 -DNDEBUG`.
Debug sets `-g`.

**Comment by nupuruttarwar** [comment]
Tested on Fedora 33, able to build successfully and start infrap4d

File: make-all.sh [hunk 2]
```diff
@@ -22,6 +23,9 @@ while true ; do
         # --prefix=<path>
         PREFIX=$2
         shift 2 ;;
+    --debug)
+        BUILD_TYPE="Debug"
```
[the remark this answers is not in the document]
Comment (ffoulkes) [reply make-all.sh:27]: The default value is RelWithDebInfo, the same as the CMakeLists.txt file.

The other developers want the ability to do debug builds. I took the minimalist approach and implemented just that. We can expand to include Release later (maybe around the time we start adding toolchain support).

**Pull request merged**